app/renderer/prefilledPaymentRequests.ts

```diff
@@ -0,0 +1,141 @@
+import { ExternalFinalKey, FinalKeyMetadata } from "app/common/runtimeTypes/storage/wallets"
+
+/**
+ * Payment request status according to the funds received
+ */
+export type FundedStatus = "pending" | "partially funded" | "funded" | "overfunded"
+
+/**
+ * Payment request status according to the expiration status
+ */
+export type ExpiredStatus = "expired" | "not expired" | "unknown"
+
+/**
+ * Payment request UI action
+ */
+export interface Action {
+  text: string,
+  onClick: Function
+}
+
+/**
+ * Computed payment request type
+ */
+export interface ComputedPaymentRequest {
+  // Expiration status
+  expiredStatus: ExpiredStatus,
+
+  // Funded status
+  fundedStatus: FundedStatus,
+
+  // Funds (in WITs)
+  funds: number,
+
+  // Address
+  address: string,
+
+  // Optional metadata
+  metadata?: FinalKeyMetadata,
+
+  // Actions
+  actions: Array<Action>
+}
+
+/**
+ * Type to represent an array of payment requests (as represented in the UI)
+ */
+export type ComputedPaymentRequests = Array<ComputedPaymentRequest>
+
+/**
+ * Function to build a computed payment request from an external final key
+ */
+export function buildComputedPaymentRequest(
+  fk: ExternalFinalKey,
+  address: string,
+  funds: number,
+  actions?: Array<Action>
+) {
+  // Check funds
+  const fundedStatus = computeFundedStatus(
+    funds,
+    fk.metadata && fk.metadata.requestedAmount
+  )
+
+  // Check expiration date
+  const expiredStatus = computeExpiredStatus(
+    fundedStatus,
+    fk.metadata && fk.metadata.expirationDate
+  )
+
+  return {
+    expiredStatus,
+    fundedStatus,
+    funds,
+    address,
+    metadata: fk.metadata,
+    actions
+  }
+}
+
+/**
+ * Function to get payment request funded status
+ */
+function computeFundedStatus(
+  funds: number,
+  requestedAmount?: number
+): FundedStatus {
+
+  return (funds === 0)
```

Comment:
IMO, we should avoid complex conditional ternaries. In a first aproach a refactor this code as:
```
  const isPending = (funds, requestedAmount) => funds === 0
  const isFunded = (funds, requestedAmount) => !requestedAmount
  const isOverfunded = (funds, requestedAmount) => funds > requestedAmount
  
  if (isPending(funds)) {
    return "pending"
  } else if (isFunded(funds, requestedAmount)){
    return "funded"
  } else if (isOverfunded(funds, requestedAmount)) {
    return "overfunded"
  } else {
    return "partially funded"
  }
```
By the other hand we can refactor it in a more functional way. How relevant is the conditional order?

Reply:
_Dude, we are now FP cool kids. Dead to the statements! F*ck the multiple exit points!_

Reply:
And that was just a proof that this week is being tough. Tonight's team dinner is well deserved.

Reply:
And now, for serious: I agree this ternary tree is damn hard to read.

Reply:
:rofl: I tried my best to avoid the multiple exit points but I agree the code looks complex

Reply:
For the sake of clarification / keeping spamming: I'd not touch the code but just add comments on what are the conditions being tested.

Reply:
I'll refactor it to make it more readable :+1: